lib/health_cards/health_card.rb

```diff
@@ -146,78 +169,90 @@ def to_json(*_args)
     # @return [Hash] A hash with the same content as the FHIR::Bundle, processed accoding
     # to SMART Health Cards framework and any constraints created by subclasses
     def strip_fhir_bundle
-      stripped_bundle = @bundle.to_hash
-      if stripped_bundle.key?('entry') && !stripped_bundle['entry'].empty?
-        entries = stripped_bundle['entry']
-        entries, @url_map = redefine_uris(entries)
-        update_elements(entries)
+      return [] unless bundle.entry
+
+      # Using this to dup the original bundle, hopefully it correctly dups everything
+      new_bundle = FHIR::Bundle.new(bundle.to_hash)
+      url_map = redefine_uris(new_bundle)
+
+      walk_resource(new_bundle) do |value, type|
+        case type
+        when 'Reference'
+          value.reference = process_reference(url_map, value)
+        when 'Resource'
+          value.id = nil
+          value.text = nil
+          value.meta = nil unless value.meta&.security
```

Comment:
It could be modified to do that, but as it exists now allow/disallow can't handle these cases. My thought was the implementation of these special cases wasn't worth it in the short term at least:

* The allow/disallow rules are based on the class, so I'd have to change it to also use ancestors (e.g. `disallow FHIR::Model, &w(id text)`
* Meta is a special case where you only nil it out if a condition is met. I'd have to add the ability to pass a proc to to allow/disallow to determine if the rule should be applied.